Fix various accessibility issues

CarlSchwan opened the pull request:
- Increase the size of input fields not fully AAA compliant but better
- Use the same design as the new vue component
- Fix user status dialog
  - Add labels where missings
  - Move emoji picker inside input field (similar to talk)
  - Fix selecting an emoji
- Fix multiselect regressions
- Small password settings improvements
- Fix button with confirmation action
- Fix some other unrelated dark theme issues
- Fix select2 focus
- Run npm lint:fix

Screenshots:

![image](https://user-images.githubusercontent.com/23653902/184106391-10c71d66-b729-478e-96ca-96ddf9f74c66.png)

![image](https://user-images.githubusercontent.com/23653902/184106564-33c07075-d5cf-4e22-9762-9d1240e86097.png)

CarlSchwan committed:
Fix accessibility of input fields

Input fields require a 44x44 pixels target size, this makes all the
input fields and button use that size.

Bonus is that now the input fields and buttons now have the same size as
the new vue button and this looks less weird than the previous state
with controls of different sizes.

See https://www.w3.org/WAI/WCAG21/Understanding/target-size.html

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

jancborchardt reviewed (changes requested):
The left/right padding needs to be adjusted, currently it’s way too little. It should be visually the same as the top/bottom padding. (Also compare to the select dropdown e.g. for language, where the left/right padding is better but actually a bit too much.)

CarlSchwan commented:
> The left/right padding needs to be adjusted, currently it’s way too little. It should be visually the same as the top/bottom padding. (Also compare to the select dropdown e.g. for language, where the left/right padding is better but actually a bit too much.)

I updated the screenshots to include the new paddings

PVince81 commented:
@CarlSchwan if this is ready for re-review, please re-request review from Jan-C instead of leaving it blocking

Pytal commented on apps/user_ldap/lib/Group_LDAP.php.rej:
```diff
@@ -0,0 +1,33 @@
+--- apps/user_ldap/lib/Group_LDAP.php
++++ apps/user_ldap/lib/Group_LDAP.php
+@@ -62,7 +62,7 @@ class Group_LDAP extends BackendUtility implements GroupInterface, IGroupLDAP, I
+        protected CappedMemoryCache $cachedGroupsByMember;
+        /** @var CappedMemoryCache<string[]> $cachedNestedGroups array of groups with gid (DN) as key */
+        protected CappedMemoryCache $cachedNestedGroups;
+-       protected GroupInterface $groupPluginManager;
++       protected GroupPluginManager $groupPluginManager;
+        protected LoggerInterface $logger;
+ 
+        /**
+@@ -243,8 +243,9 @@ class Group_LDAP extends BackendUtility implements GroupInterface, IGroupLDAP, I
+         * @psalm-param array<string, bool> $seen List of DN that have already been processed.
+         * @throws ServerNotAvailableException
+         */
+-       private function _groupMembers(string $dnGroup, array &$seen = []): array {
++       private function _groupMembers(string $dnGroup, array $seen = [], bool &$recursive = false): array {
+                if (isset($seen[$dnGroup])) {
++                       $recursive = true;
+                        return [];
+                }
+                $seen[$dnGroup] = true;
+@@ -317,7 +318,9 @@ class Group_LDAP extends BackendUtility implements GroupInterface, IGroupLDAP, I
+                        unset($allMembers[$index]);
+                }
+ 
+-               $this->access->connection->writeToCache($cacheKey, $allMembers);
++               if (!$recursive) {
++                       $this->access->connection->writeToCache($cacheKey, $allMembers);
++               }
+ 
+                if (isset($attemptedLdapMatchingRuleInChain)
+ 
```
This file should be removed

Pytal reviewed (dismissed):
Other than the comment below

jancborchardt reviewed (approved):
Looks good now! :)

jancborchardt commented on core/css/inputs.scss:
```diff
@@ -51,7 +51,7 @@ textarea,
 	margin: 3px 3px 3px 0;
-	padding: 7px 6px;
+	padding: 7px 14px;
 	font-size: 13px;
```
Btw @CarlSchwan while upping the size, we could also adjust this to be the default font size (15px), then it would look less strange and be legible too. As per https://github.com/nextcloud/nextcloud-vue/issues/2172

CarlSchwan commented:
Probably also best to wait for https://github.com/nextcloud/nextcloud-vue/pull/2868 to get merged to adapt the style/size as much as possible to the new view component

PVince81 commented:
@CarlSchwan still ready to merge then ? it's green!

CarlSchwan committed:
Adapt to new Nextcloud vue style

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

CarlSchwan commented:
Adapted to the new style of Nextcloud vue components and change the font size to 15px

![image](https://user-images.githubusercontent.com/23653902/184105585-6a923d49-0ee4-44eb-9a40-332b2f0d5985.png)

nickvergessen commented:
Somehow this is limiting the width of selects by default very much.

Talk code in both screenshots is unmodified

Before | After
---|---
![Bildschirmfoto vom 2022-08-11 12-12-17](https://user-images.githubusercontent.com/213943/184112055-af2c20c4-f6a3-4a3c-b43f-caff08acfe01.png) | ![Bildschirmfoto vom 2022-08-11 12-12-28](https://user-images.githubusercontent.com/213943/184112060-d605b41b-0a01-44aa-8490-88200e0f75e8.png)

nickvergessen commented on core/css/inputs.scss:
```diff
@@ -174,13 +173,50 @@ input {
+}
+
+
+/* 'Click' inputs */
 button:not(.button-vue), .button,
 input[type='button'],
 input[type='submit'],
 input[type='reset'] {
```
you effectively removed select from this so the size doesn't grow to `width:auto` anymore causing the outlined issue in talks admin settings (and many other)

CarlSchwan replied:
Should be fixed now

nickvergessen replied:
Yes, fixed

nickvergessen commented:
This brings in various regressions all over the place which I kind a dislike on the last day.

### Talk participant search
![Bildschirmfoto vom 2022-08-12 08-30-16](https://user-images.githubusercontent.com/213943/184297865-20cd692b-d072-47f1-9e7a-4b32709e51d2.png)

### Talk room filter pushes off the + button
![Bildschirmfoto vom 2022-08-12 08-31-02](https://user-images.githubusercontent.com/213943/184297870-4e0ad541-f073-469d-b22a-b3ff53c2e43f.png)

### Multiselects (Vue component) get white background aspects?!
Before | After
---|---
![Bildschirmfoto vom 2022-08-12 08-33-49](https://user-images.githubusercontent.com/213943/184297872-c38fa4b0-0780-4cc0-a570-cc0101e6785f.png) | ![Bildschirmfoto vom 2022-08-12 08-34-01](https://user-images.githubusercontent.com/213943/184297873-8f943980-8928-4019-9fad-cfe9c20c559b.png)
![Bildschirmfoto vom 2022-08-12 08-36-17](https://user-images.githubusercontent.com/213943/184298095-fe666910-d16d-4812-8b70-6cc309958fb2.png) | ![Bildschirmfoto vom 2022-08-12 08-36-06](https://user-images.githubusercontent.com/213943/184298099-4e8e1b6b-d131-4987-8379-3b72c7df6545.png)

While I get the point of making all inputs look the same, last minutes enforcements like this are a bit painful.

nickvergessen commented:
### Elements get double borders
![Bildschirmfoto vom 2022-08-12 08-38-57](https://user-images.githubusercontent.com/213943/184299190-1fdea8b6-1b6c-4680-a593-16e5fd2f2b30.png)
![Bildschirmfoto vom 2022-08-12 08-44-04](https://user-images.githubusercontent.com/213943/184299199-7b83c921-90e9-49ac-8cf1-f01b5d9558c1.png)

### Share input is white
![Bildschirmfoto vom 2022-08-12 08-39-37](https://user-images.githubusercontent.com/213943/184299191-c46f9c3b-9651-4e56-9cb8-b8eaf788771f.png)

### Combined submit button looks weird
![Bildschirmfoto vom 2022-08-12 08-41-02](https://user-images.githubusercontent.com/213943/184299194-1d73337f-0cbe-4fb9-ad11-883e9d3deb2e.png)

### Contacts menu button stretched

Before | After
---|---
![Bildschirmfoto vom 2022-08-12 08-42-56](https://user-images.githubusercontent.com/213943/184299196-a47d8dc9-c410-4585-96c9-163d5585da59.png) | ![Bildschirmfoto vom 2022-08-12 08-43-07](https://user-images.githubusercontent.com/213943/184299198-9bb8991d-6647-4b07-a8f1-cf0e677a7c1b.png)

nickvergessen reviewed (changes requested):
See above

PVince81 commented:
@CarlSchwan for now we're first aiming for AA, not AAA, so merging this isn't critical if it's only about target size

CarlSchwan committed:
Various fixes

- Fix user status dialog
  - Add label where missing
  - Move emoji picker inside input field (similar to talk)
  - Fix selecting an emoji
- Fix multiselect
- Fix button with confirmation action
- Fix some other unrelated dark theme issues
- Fix select2 focus
- Run npm lint:fix

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

CarlSchwan commented:
From the list of issues reported by @nickvergessen only the search field component issue remains and this should probably just be ported to the proper TextField component with a leading icons

CarlSchwan committed:
More dark theme fixes

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

CarlSchwan committed:
Make it possible to load theme/<themename>/dist

This fix a regression with the theming api following the change in
webpack bundling introduced in 24

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

CarlSchwan committed:
Fix rebasing issues and improve profile settings page a bit more

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

PVince81 commented:
oops, I merged @Pytal's #33217 and now there's conflicts again with your changes :-/

Pytal commented:
🤔 looks like we just need to adapt the styles in https://github.com/nextcloud/server/blob/master/apps/settings/src/components/PersonalInfo/shared/AccountPropertySection.vue#L192-L265 @CarlSchwan

CarlSchwan commented:
> oops, I merged @Pytal's #33217 and now there's conflicts again with your changes :-/

rebased :) let's hope it is the last time

nickvergessen commented:
> Conflicting files
dist/core-main.js.map 

Merged by @CarlSchwan 
https://github.com/nextcloud/server/commit/aa150b9f85a2543fdbac8e73b5e6f8bb39f125f4

Shame on him!

miaulalala reviewed (approved):
:elephant:

skjnldsv reviewed (approved):
:elephant: :elephant:

PVince81 reviewed (approved):
:+1: (we got a nice tour of this yesterday)

skjnldsv commented:
/rebase
